zerver/views/auth.py

```diff
@@ -617,7 +619,12 @@ def start_social_signup(
             return config_error(request, "saml")
         extra_url_params = {"idp": extra_arg}
     return oauth_redirect_to_root(
-        request, backend_url, "social", is_signup=True, extra_url_params=extra_url_params
+        request,
+        backend_url,
+        "social",
+        is_signup=True,
+        extra_url_params=extra_url_params,
+        next=next,
```

Comment:
Shouldn't the `next` param get just get picked up by `oauth_redirect_to_root` without needing this, because the function has the `has_request_variables` decorator? (That's how it picks up the `mobile_flow_otp` and other params iirc)

Reply:
Oh! I wasn't aware that this was possible.

Reply:
Doesn't seems to be working without this change.

Reply:
@amanagr Hmm, I'm not sure why :thinking: Just tested with your PR and any explicit `next` plumbing removed in `start_social_login, put `breakpoint()` at the start of `oauth_redirect_to_root` and if I go to e.g. `http://localhost:9991/accounts/login/social/github?next=/test`, it gets correctly picked up in the funciton:
```
> /srv/zulip/zerver/views/auth.py(510)oauth_redirect_to_root()
-> main_site_uri = settings.ROOT_DOMAIN_URI + url
(Pdb) print(next)
/test
(Pdb) c
2022-06-08 09:50:42.027 INFO [zr] 172.17.0.1      GET     302 35.3s (+start: 9ms) /accounts/login/social/github (unauth@zulip via Mozilla)
2022-06-08 09:50:42.029 INFO [django.server] "GET /accounts/login/social/github?next=/test HTTP/1.1" 302 0
```

Reply:
Ok, I figured out what I was  doing wrong. This new set of changes should fix this. Please review agian!